src/simulation_lifecycle/simulation.c

```diff
@@ -89,3 +93,103 @@ int write_sim_config(const cJSON *simulation_config, char *config_json) {
     }
     return write_data_to_file(cJSON_GetStringValue(config_output_path), config_json);
 }
+
+int run_sim(const cJSON *simulation_config){
+
+    /* Get model name - e.g. 2_2_agent_sir_config */
+    cJSON *model = cJSON_GetObjectItemCaseSensitive(simulation_config, SIM_MODEL_ID);
+
+    /* Check that model ID is provided using a valid format. */
+    if (model == NULL || !cJSON_IsString(model)) {
+        return SIM_MODEL_SELECTION_INVALID;
+    }
+
+    /* Check that the model exists. */
+    char *model_path = concat(SIM_MODEL_LIBRARY, cJSON_GetStringValue(model));
+    int model_found = file_exists(model_path);
```

Comment:
Same here, you should use the `executable_exists` function

Reply:
Change made.